public/index.js

```diff
@@ -0,0 +1,368 @@
+
+    this.averageRatingElement.textContent = `${this.averageRating.toFixed(
+      1
+    )} / 5`;
+  }
+
+  /**
+   * Adds a review to the reviews array and re-renders the review list
+   * @param {{review: string, rating: number}} review
+   * @param {'start' | 'end'} location
+   */
+  addReview(review, location = "end") {
+    if (
+      typeof review.rating !== "number" ||
+      review.rating < 1 ||
+      review.rating > 5
+    ) {
+      throw new Error(
+        "Invalid rating. Rating must be a number between 1 and 5."
+      );
+    }
+
+    if (location === "start") {
+      this.reviews.unshift(review);
+    } else if (location === "end") {
+      this.reviews.push(review);
+    } else {
+      throw new Error("Invalid location provided");
+    }
+
+    this.storageAdapter.save(this.reviews);
+    this.updateAverageRating();
+    this.render();
+  }
+
+  /**
+   * Removes a review from the reviews array by index and re-renders the review list
+   * @param {number} index
+   */
+  removeReviewByIndex(index) {
+    if (index < 0 || index >= this.reviews.length) {
+      throw new Error("Index out of bounds");
+    }
+
+    this.reviews.splice(index, 1);
+
+    this.storageAdapter.save(this.reviews);
+    this.updateAverageRating();
+    this.render();
+  }
+
+  /**
+   * Removes a review from the reviews array by reference and re-renders the review list
+   * @param {{review: string, rating: number}} review
+   */
+  removeReviewByReference(review) {
+    const index = this.reviews.indexOf(review);
+    if (index === -1) {
+      throw new Error("Review not found");
+    }
+    this.removeReviewByIndex(index);
+  }
+
+  /**
+   * Renders the review list
+   */
+  render() {
+    this.element.innerHTML = "";
+
+    this.reviews.forEach((review) => {
+      const reviewCard = new ReviewCard(review, this);
+      this.element.appendChild(reviewCard.element);
+    });
+  }
+}
+
+// Create a new ReviewList instance and mount it to the DOM
+const averageRatingElement = document.getElementById("average-rating");
+const reviewList = new ReviewList(reviews, averageRatingElement);
+const reviewListMountElement = document.getElementById("review-list");
+reviewListMountElement.appendChild(reviewList.element);
+
+// Create a new RatingInput instance and mount it to the DOM
+const reviewFormRatingComponentElement = document.getElementById(
+  "review-form__rating-component"
+);
+const ratingInput = new RatingInput();
+reviewFormRatingComponentElement.appendChild(ratingInput.element);
+
+/**
+ * Handles form submission for adding a new review
+ * @param {Event} event
+ */
+const reviewFormSubmitHandler = (event) => {
+  event.preventDefault();
+  event.stopPropagation();
+
+  const form = event.target;
+  const formData = new FormData(form);
+
+  const formValues = {};
+  formData.forEach((value, key) => {
+    formValues[key] = value;
+  });
+
+  const review = {
+    review: formValues.review,
+    rating: ratingInput.ratingValue,
```

Comment:
Postoji neki razlog zasto ratingValue ne koristis iz form state-a, vec iz ratingInputa?
Works, ali imaj na umu da je dobro imati što manje tzv. source of truth.

Reply:
Inicijalno sam to napravio kada sam imao number input pa sam maknuo jer sam zbog nekog razloga mislio da mi nece raditi sa RatingInput ali ipak radi.

Fixano u novom commitu.